### PR TITLE
only runs `yarn install`

### PR DESCRIPTION
**Pull request by joshzarrabi**
- stops getting the `yarn` binary -- moved to `yarn` buildpack
- remove `yarn start logic` -- moved to `yarn start` buildpack

Signed-off-by: Josh Zarrabi <jzarrabi@vmware.com>

### CODE REVIEW

File: build.go [hunk 1]
```diff
@@ -119,25 +81,18 @@ func Build(dependencyService DependencyService, cacheMatcher CacheMatcher, insta
 			if err != nil {
 				return packit.BuildResult{}, err
 			}
-
 			err = os.Symlink(filepath.Join(modulesLayer.Path, "node_modules"), filepath.Join(context.WorkingDir, "node_modules"))
 			if err != nil {
+				// not tested
```
Comment (arjun024) [inline build.go:86]: Is this comment left here intentionally?

File: install_process_test.go [hunk 2]
```diff
@@ -180,7 +175,6 @@ func testInstallProcess(t *testing.T, context spec.G, it spec.S) {
 				"get",
 				"yarn-offline-mirror",
 			}))
-			Expect(executions[0].Env).To(ContainElement(MatchRegexp(fmt.Sprintf(`^PATH=.*:%s/bin:node_modules/.bin$`, yarnLayerPath))))
```
Comment (arjun024) [inline install_process_test.go:178]:
```suggestion
Expect(executions[0].Env).To(ContainElement(MatchRegexp(`^PATH=.*:node_modules/.bin$`)))
```
Don't we still need to assert that "node_modules/.bin" is added to path?

File: install_process_test.go [hunk 3]
```diff
@@ -189,15 +183,12 @@ func testInstallProcess(t *testing.T, context spec.G, it spec.S) {
 				"--modules-folder",
 				filepath.Join(modulesLayerPath, "node_modules"),
 			}))
-			Expect(executions[1].Env).To(ContainElement(MatchRegexp(`^PATH=.*:node_modules/.bin$`)))
```
Comment (arjun024) [inline install_process_test.go:186]:
```suggestion
Expect(executions[1].Env).To(ContainElement(MatchRegexp(`^PATH=.*:node_modules/.bin$`))
```
Isn't this still relevant?

File: install_process_test.go [hunk 4]
```diff
@@ -225,7 +216,6 @@ func testInstallProcess(t *testing.T, context spec.G, it spec.S) {
 					"get",
 					"yarn-offline-mirror",
 				}))
-				Expect(executions[0].Env).To(ContainElement(MatchRegexp(fmt.Sprintf(`^PATH=.*:%s/bin:node_modules/.bin$`, yarnLayerPath))))
```
Comment (arjun024) [inline install_process_test.go:228]: Same query as above with `node_modules/.bin`

File: install_process_test.go [hunk 5]
```diff
@@ -235,7 +225,6 @@ func testInstallProcess(t *testing.T, context spec.G, it spec.S) {
 					"--modules-folder",
 					filepath.Join(modulesLayerPath, "node_modules"),
 				}))
-				Expect(executions[1].Env).To(ContainElement(MatchRegexp(fmt.Sprintf(`^PATH=.*:%s/bin:node_modules/.bin$`, yarnLayerPath))))
```
Comment (arjun024) [inline install_process_test.go:238]: Same query as above with node_modules/.bin

File: integration/vendored_test.go [hunk 6]
```diff
@@ -56,19 +53,26 @@ func testVendored(t *testing.T, context spec.G, it spec.S) {
...
+					buildpackOfflineURI,
+					buildPlanURI,
+				).
 				WithNetwork("none").
 				Execute(name, source)
 			Expect(err).NotTo(HaveOccurred())
 
-			container, err = docker.Container.Run.Execute(image.ID)
+			container, err = docker.Container.Run.
+				WithCommand("ls -alR /layers/paketo-buildpacks_yarn-install/modules/node_modules").
```
Comment (arjun024) [inline integration/vendored_test.go:67]: Can we avoid hardcoding `paketo-buildpacks_yarn-install` and instead derive the layer name from the buildpack.toml info?

File: integration/caching_test.go [hunk 7]
```diff
@@ -58,60 +58,64 @@ func testCaching(t *testing.T, context spec.G, it spec.S) {
...
-				Expect(secondImage.Buildpacks[1].Key).To(Equal(buildpackInfo.Buildpack.ID))
-				Expect(secondImage.Buildpacks[1].Layers).To(HaveKey("yarn"))
-				Expect(secondImage.Buildpacks[1].Layers).To(HaveKey("modules"))
+				Expect(secondImage.Buildpacks).To(HaveLen(4))
+				Expect(secondImage.Buildpacks[2].Key).To(Equal(buildpackInfo.Buildpack.ID))
+				Expect(secondImage.Buildpacks[2].Layers).To(HaveKey("modules"))
 
-				container, err = docker.Container.Run.Execute(secondImage.ID)
+				container, err = docker.Container.Run.
+					WithCommand("ls -alR /layers/paketo-buildpacks_yarn-install/modules/node_modules").
```
Comment (arjun024) [inline integration/caching_test.go:100]: Can we avoid hardcoding paketo-buildpacks_yarn-install and instead derive the layer name from the buildpack.toml info?

File: integration/simple_app_test.go [hunk 8]
```diff
@@ -56,19 +53,26 @@ func testSimpleApp(t *testing.T, context spec.G, it spec.S) {
...
+					buildpackURI,
+					buildPlanURI,
+				).
 				WithNoPull().
 				Execute(name, source)
 			Expect(err).NotTo(HaveOccurred())
 
-			container, err = docker.Container.Run.Execute(image.ID)
+			container, err = docker.Container.Run.
+				WithCommand("ls -alR /layers/paketo-buildpacks_yarn-install/modules/node_modules").
```
Comment (arjun024) [inline integration/simple_app_test.go:67]: Can we avoid hardcoding paketo-buildpacks_yarn-install and instead derive the layer name from the buildpack.toml info?

File: build.go (hunk 1 above)
Comment (joshzarrabi) [reply build.go:86]: Yeah! We couldn't figure out an easy way to hit the `if` case in our test, and in those circumstances I like to be transparent.

File: install_process_test.go (hunk 2 above)
Comment (joshzarrabi) [reply install_process_test.go:178]: good call! Not sure why we deleted these, must have been an accident!

**Review by arjun024 — APPROVED** [review]

**Pull request merged**